youtube_dl/extractor/nhk.py

```diff
@@ -29,48 +26,54 @@ def _list_episodes(self, lang, m_type, m_id):
     def _parse_episode_json(self, episode, lang, is_video):
         title = episode.get('sub_title_clean') or episode['sub_title']
 
-        episode_id = episode.get('pgm_id') + episode.get('pgm_no')
-
-        series = self._get_clean_field(episode, 'title')
-
-        thumbnails = []
-        for s, w, h in [('', 640, 360), ('_l', 1280, 720)]:
-            img_path = episode.get('image' + s)
-            if not img_path:
-                continue
-            thumbnails.append({
-                'id': '%dp' % h,
-                'height': h,
-                'width': w,
-                'url': 'https://www3.nhk.or.jp' + img_path,
-            })
-
-        info = {
-            'id': episode_id + '-' + lang,
-            'title': '%s - %s' % (series, title) if series and title else title,
-            'description': self._get_clean_field(episode, 'description'),
-            'thumbnails': thumbnails,
-            'series': series,
-            'episode': title,
-        }
-
-        if is_video:
-            info.update({
-                '_type': 'url_transparent',
-                'ie_key': 'Piksel',
-                'url': 'https://player.piksel.com/v/refid/nhkworld/prefid/' + episode['vod_id'],
-            })
-        else:
-            audio = episode['audio']
-            audio_path = audio['audio']
-            info['formats'] = self._extract_m3u8_formats(
-                'https://nhkworld-vh.akamaihd.net/i%s/master.m3u8' % audio_path,
-                episode_id, 'm4a', entry_protocol='m3u8_native',
-                m3u8_id='hls', fatal=False)
-            for f in info['formats']:
-                f['language'] = lang
+        pgm_id = episode.get('pgm_id')
+        pgm_no = episode.get('pgm_no')
 
-        return info
+        if (pgm_id == None or pgm_no == None):
+            raise ExtractorError('Cannot download episode: %s' % str(episode), expected=True)
```

Comment:
use `compat_str` instead of `str`, and I'm not what you would call `str` on a `dict`.